### DISCUSSION TITLE
some items are unclear

### DISCUSSION THREAD

**Discussion by melvincarvalho** [renostr/nrps#5] [Q&A]
- where to raise issues?  there's no issues on the nrp
- what does ots field do?
- how to do on-chain commitments, because ots is proof of publication only
- example of how to interact with RGB is not there
- what is the speed difference in parsing binary and json (I have JSON.parse running 3m/s, and 500k/s for nostr events w/ envelope) - is it really 25x for binary

**Comment by dr-orlovsky** [comment]
> where to raise issues? there's no issues on the nrp

I took an example from BIPs where there is no issues and just PRs. There is also Discussions to discuss the questions like in your list https://github.com/orgs/renostr/discussions

> what does ots field do?

OpenTimeStamps - the same as in the NOP-1

> how to do on-chain commitments, because ots is proof of publication only

Not sure I understand the question

> example of how to interact with RGB is not there

it will require a dedicated standard

> what is the speed difference in parsing binary and json (I have JSON.parse running 3m/s, and 500k/s for nostr events w/ envelope) - is it really 25x for binary

This is an indication I was given by @fiatjaf for Go implementation. Later he provided info that on JavaScript JSON is faster (however I am not sure where `ByteArray`/`Blob` were used).

**Comment by melvincarvalho** [comment]
All good!

> OpenTimeStamps - the same as in the NOP-1
> Not sure I understand the question
> it will require a dedicated standard

I'm interested in the mechanism to bring smart contracts to renostr and nostr.  I guess right now it's using OTS as proof of publication.

> Merkelized commitments

The readme says this, I suppose you mean by this OTS, proof of publication, then

> This is an indication I was given by @fiatjaf for Go implementation. Later he provided info that on JavaScript JSON is faster (however I am not sure where ByteArray/Blob were used).

OK it would be good to have actual benchmarks.  I still have hope that nostr and renostr can share use cases.

**Comment by dr-orlovsky** [comment]
I have converted this issue into discussions Q&A section; hope you are ok with this.

> I'm interested in the mechanism to bring smart contracts to renostr and nostr. I guess right now it's using OTS as proof of publication.

"Smart contract" is a broad term. If you can do threshold signatures with a dynamic participant set and get DAOs operating in nostr, wouldn't it be a smart contract?

> The readme says this, I suppose you mean by this OTS, proof of publication, then

No, what I meant was the fact that the hash (over which a signature is performed) should not be created from the plan serialized even but must be built via merklization of independent hashes of the even fields.
